editor/plugins/texture_editor_plugin.cpp

```diff
@@ -124,19 +149,42 @@ void TexturePreview::_update_metadata_label_text() {
 }
 
 TexturePreview::TexturePreview(Ref<Texture2D> p_texture, bool p_show_metadata) {
+	set_custom_minimum_size(Size2(0.0, 256.0) * EDSCALE);
+
+	bg_rect = memnew(ColorRect);
+	bg_rect->set_anchors_preset(PRESET_FULL_RECT);
+	add_child(bg_rect);
+
+	margin_container = memnew(MarginContainer);
+	margin_container->set_anchors_preset(PRESET_FULL_RECT);
+	add_child(margin_container);
+
+	centering_container = memnew(AspectRatioContainer);
+	centering_container->set_anchors_preset(PRESET_FULL_RECT);
+	margin_container->add_child(centering_container);
+
 	checkerboard = memnew(TextureRect);
+	checkerboard->set_anchors_preset(PRESET_FULL_RECT);
+	checkerboard->set_expand_mode(TextureRect::EXPAND_IGNORE_SIZE);
 	checkerboard->set_stretch_mode(TextureRect::STRETCH_TILE);
 	checkerboard->set_texture_repeat(CanvasItem::TEXTURE_REPEAT_ENABLED);
-	checkerboard->set_custom_minimum_size(Size2(0.0, 256.0) * EDSCALE);
-	add_child(checkerboard);
+	centering_container->add_child(checkerboard);
 
 	texture_display = memnew(TextureRect);
 	texture_display->set_texture_filter(TEXTURE_FILTER_NEAREST_WITH_MIPMAPS);
 	texture_display->set_texture(p_texture);
-	texture_display->set_anchors_preset(TextureRect::PRESET_FULL_RECT);
-	texture_display->set_stretch_mode(TextureRect::STRETCH_KEEP_ASPECT_CENTERED);
 	texture_display->set_expand_mode(TextureRect::EXPAND_IGNORE_SIZE);
-	add_child(texture_display);
+	texture_display->set_anchors_preset(PRESET_FULL_RECT);
+	centering_container->add_child(texture_display);
+
+	texture_display->connect(SceneStringName(draw), callable_mp(this, &TexturePreview::_draw_outline));
+
+	set_texture_display_outline_width(outline_width);
+
+	if (p_texture.is_valid()) {
```

Comment:
Oh, please also add similar check for `_update_metadata_label_text();` down below (which was added by... me :upside_down_face:), seems like it can be created with null `p_texture`.

Reply:
Sure! Done.